x/dex/keeper/grpc_query_user_deposits.go

```diff
@@ -56,6 +56,10 @@ func (k Keeper) UserDepositsAll(
 					Fee:             poolMetadata.Fee,
 				}
 
+				if req.IncludePoolData {
+					depositRecord = k.addPoolData(ctx, depositRecord)
```

Comment:
looks a bit confusing to me, semantic looks like `append` function, may be it because you return the same pointer

Reply:
you mean that the `depositRecord = ` part could be removed since depositRecord is passed as a pointer and mutations are applied to the record without an additional assignment? if yes, I agree